Sources/Utils.php

```diff
@@ -2264,6 +2277,27 @@ public static function getCallable(mixed $input, ?bool $ignore_errors = null): m
 		return $callable;
 	}
 
+	/**
+	 * Backward compatibilty wrapper for the smf_json_decode() method.
+	 *
+	 * @deprecated 3.0 - Only exists for $smcFunc backwards wrapper.
+	 * @param mixed $json The string to decode.
+	 * @param bool $associative Whether to force JSON objects to be returned as
+	 *    associative arrays. SMF nearly always wants this to be true, but for
+	 *    the sake of consistency with json_decode(), the default is false.
+	 * @param bool $should_log Whether to log errors. Default: true.
+	 * @return mixed The decoded data.
+	 */
+	public static function backcompat_smf_json_decode(mixed $json, bool $associative = false, bool $should_log = true): mixed
+	{
+		// In older versions, we accepted a mixed $json and would return if it was not a string.
+		if (empty($json) || !is_string($json)) {
+			return $json;
+		}
+
+		return Utils::jsonDecode($json, $associative, 512, 0, $should_log);
+	}
+
```

Comment:
It would be better to paste the contents of this method into the `smf_json_decode()` function in Subs-Compat.php, and then have `$smcFunc['json_decode']` point to that.

Reply:
I think It is correct now on how we would expect it to work.

Interesting that `$smcFunc['json_decode']` was mapping to `Utils::jsonDecode` instead of the `smf_json_decode`.

Reply:
> Interesting that `$smcFunc['json_decode']` was mapping to `Utils::jsonDecode` instead of the `smf_json_decode`.

That's due to the history and evolution of the code. In my original OOP refactoring, `$smcFunc['json_decode']` and `smf_json_decode` were both just aliases of `Utils::jsonDecode` that were `eval`'d into existence at runtime. But in @Tyrsson's refactoring of BackwardCompatibilty, `smf_json_decode` became a real function again. As long as the signatures of all three were the same, it made sense for `$smcFunc['json_decode']` to continue to be a direct alias of `Utils::jsonDecode` rather than detouring through `smf_json_decode`. But now that the signature of `Utils::jsonDecode` has changed, making `$smcFunc['json_decode']` an alias of `smf_json_decode` makes the most sense.